src/main/webapp/app/exercises/programming/manage/programming-exercise-detail.component.ts

```diff
@@ -206,6 +207,12 @@ export class ProgrammingExerciseDetailComponent implements OnInit, OnDestroy {
                     content: 'Deleted a programming exercise',
                 });
                 this.dialogErrorSource.next('');
+
+                if (!this.isExamExercise) {
+                    this.router.navigateByUrl(`/course-management/${this.courseId}/exercises`);
+                } else {
+                    this.router.navigateByUrl(`/course-management/${this.courseId}/exams/${this.programmingExercise.exerciseGroup?.exam?.id}/exercise-groups`);
+                }
```

Comment:
It seems like similar code can be found here as well: https://github.com/ls1intum/Artemis/pull/3798/files#diff-a20549d907c835b47b8f21b0eb5c784240fa472267fcab0d6391d9dcea756f51R99-R104

What do you think of extracting it to navigation utils?